lib/ansible/modules/database/postgresql/postgresql_query.py

```diff
@@ -0,0 +1,330 @@
+    login_password: ceec4eif7ya
+    query: SELECT * FROM some_table WHERE a_column=%(a_value)s AND b_column=%(b_value)s
+    positional_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+
+
+# Run queries from a '.sql' file
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: "{{playbook_dir}}/scripts/my_sql_query_file.sql"
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+
+# Run queries from a '.sql' file and assign result in a fact available for the
+# rest of the ansible runtime. Query inside scripts may contain 'named_args'
+# or 'positional_args'.
+- postgresql_query:
+    db: acme
+    login_user: django
+    login_password: ceec4eif7ya
+    query: SELECT * FROM some_table WHERE a_column=%(a_value)s AND b_column=%(b_value)s
+    query: /path/to/my/script.sql
+    named_args:
+      a_value: "positional string value 1"
+      b_value: "positional string value 2"
+    fact: my_key
+'''
+
+RETURN = '''
+query_results:
+    description: List of dictionaries in column:value form representing returned rows.
+    returned: changed
+    type: list
+    sample: [{"Column": "Value1"},{"Column": "Value2"}]
+row_count:
+    description: number of affected rows by query, if applicable
+    returned: changed
+    type: int
+    sample: 5
+'''
+
+import traceback
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.basic import AnsibleModule, BOOLEANS
+import ansible.module_utils.postgres as pgutils
+
+
+def main():
+    argument_spec = pgutils.postgres_common_argument_spec()
+
+    argument_spec.update(dict(
+        db=dict(default=None),
+        query=dict(type="str"),
+        autocommit=dict(type='bool', default=False, choices=BOOLEANS),
+        positional_args=dict(type="list"),
+        named_args=dict(type="dict"),
+        fact=dict(default=None),
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ["positional_args", "named_args"]
+        ],
+        supports_check_mode=False,
+    )
+
+    changed = False
+    psycopg2 = pgutils.psycopg2
+    ansible_facts = {}
+    query_results = []
+    rowcount = 0
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port",
+        "db": "database",
+        "ssl_mode": "sslmode",
+        "ssl_rootcert": "sslrootcert"
+    }
+
+    kw = dict((params_map[k], v) for (k, v) in iteritems(module.params)
+              if k in params_map and v != '' and v is not None)
+
+    # If a login_unix_socket is specified, incorporate it here.
+    is_localhost = "host" not in kw or kw["host"] == "" or kw["host"] == "localhost"
```

Comment:
This doesn't seems right:
- `kw["host"]` can not be equal to empty string
- What if user specified `127.0.0.1` ?

`login_unix_socket` parameter should be removed because `host` could be used instead but then documentation fragment and `module_utils.postgres` could not be used :-/ I propose to keep `login_unix_socket` remove `is_localhost` and add `['host', 'login_unix_socket']` to `mutually_exclusive`.

[Connection URIs](https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING) should be used, it should be done for all PostgreSQL modules. Something to do for Ansible 2.5 :)

Reply:
> I propose to keep login_unix_socket remove is_localhost and add ['host', 'login_unix_socket'] to mutually_exclusive.
must be done for this module. Other improvements should be solved in another pull request for all PostgreSQL modules.

Reply:
Your solution is much neater. I removed the is_localhost thing i shamely copied from other prostgres modules...

Yes URI, one day... Would be saner...